src/common/hobject.h

```diff
@@ -344,14 +372,11 @@ CEPH_HASH_NAMESPACE_END
 
 ostream& operator<<(ostream& out, const ghobject_t& o);
 
-WRITE_EQ_OPERATORS_3(ghobject_t, hobj, shard_id, generation)
-// sort ghobject_t's by <hobj, shard_id, generation> 
-// 
-// Two objects which differ by generation are more related than
-// two objects of the same generation which differ by shard.
-// 
-WRITE_CMP_OPERATORS_3(ghobject_t,
-		      hobj,
+WRITE_EQ_OPERATORS_4(ghobject_t, max, shard_id, hobj, generation)
+
+WRITE_CMP_OPERATORS_4(ghobject_t,
+		      max,
```

Comment:
Do WRITE_EQ_OPERATORS_4 and WRITE_CMP_OPERATORS_4  without adding a bool max to ghobject_t.  These macros should still work if you have hobj.max with friend class.